src/ape/api/providers.py

```diff
@@ -1316,6 +1345,76 @@ def _find_txn_by_account_and_nonce(
+            else new_block_timeout
+        )
+
+        if required_confirmations is None:
+            required_confirmations = self.network_confirmations
+
+        last_yielded_height = None
+
+        while True:
+            if stop_block is not None and last_yielded_height is not None and last_yielded_height >= stop_block:
+                break
+            changes = self.web3.eth.get_filter_changes(filter.filter_id)
+            for new_block_hash in changes:
+                block = self.web3.eth.get_block(new_block_hash)
+                confirmed_block_number = block.number - required_confirmations
+                if last_yielded_height and confirmed_block_number < last_yielded_height:
+                    num_blocks_behind = last_yielded_height - confirmed_block_number
+                    if num_blocks_behind > required_confirmations:
+                        logger.error(f"{num_blocks_behind} Block reorganization detected. Try adjusting the required network confirmations")
+                    else:
+                        logger.warning(f"{num_blocks_behind} Block reorganization detected. Reorg is within the required network confirmations")
+                        last_yielded_height = confirmed_block_number
+                        continue
+                confirmed_block = self.web3.eth.get_block(confirmed_block_number)
+                start_time = time.time()
+                while confirmed_block is None:
+                    if time.time() - start_time > timeout:
+                        raise RuntimeError(
+                            f"Timed out waiting for block {confirmed_block_number} to be available."
+                        )
+                    logger.warning(
+                        f"Block {confirmed_block_number} not found. Waiting 1 seconds."
+                    )
+                    time.sleep(1)
+                    confirmed_block = self.web3.eth.get_block(confirmed_block_number)
+                if last_yielded_height and confirmed_block.number < last_yielded_height:
+                    num_blocks_behind = last_yielded_height - confirmed_block.number
+                    logger.error(f"{num_blocks_behind} Block reorganization detected. Try adjusting the required network confirmations")
+                last_yielded_height = confirmed_block.number
+                yield confirmed_block
+
+    def poll_logs(
+            self,
+            stop_block: Optional[int] = None,
+            address: Optional[AddressType] = None,
+            topics: Optional[List[Union[str, List[str]]]] = None,
+            required_confirmations: Optional[int] = None,
+            new_block_timeout: Optional[int] = None) -> Iterator[ContractLog]:
+        required_confirmations = (
+            required_confirmations or self.provider.network.required_confirmations
+        )
+        for block in self.poll_blocks(stop_block, required_confirmations, new_block_timeout):
+            yield from self.web3.eth.get_logs({
```

Comment:
```suggestion
            for log in self.web3.eth.get_logs({
            ...
            yield ContractLog.parse_obj(log)
```

something like that?